lib/srv/alpnproxy/kube.go

```diff
@@ -83,13 +88,22 @@ type KubeMiddleware struct {
 	certs KubeClientCerts
 }
 
+type KubeMiddlewareConfig struct {
+	Certs        KubeClientCerts
+	CertReissuer KubeCertReissuer
+	Headless     bool
```

Comment:
nit. Instead of `Headless`, maybe we can just pass in `ReissueClientWait` directly?

Reply:
Yep, I've started with that approach but then switched to `headless` flag. I think it looks a bit more clean with the flag instead of caller deciding on timeout directly and also it can be beneficial for it to know if it's running in headless mode - there might be some other behavior changes in that case in future. We of course can change it if needed for something, but for now I'll leave it like this.